src/features/navbar/mobile/NavbarMobile.tsx

```diff
@@ -0,0 +1,56 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import { Menu } from 'lucide-react';
+import { NavbarMobileProps } from './model';
+import { NavItem } from '@/features';
+import { cn } from '@/lib/utils';
+
+export function NavbarMobile(props: NavbarMobileProps) {
+  const { navigation } = props;
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const menuStyles = isMenuOpen ? 'h-screen z-10' : 'h-[0rem]';
+
+  function handleOpenMenu() {
+    setIsMenuOpen(!isMenuOpen);
+  }
+
+  document.body.style.overflow = isMenuOpen ? 'hidden' : 'auto';
+  useEffect(() => {
+    const htmlEl = document.getElementById('html');
+    const bodyEl = document.getElementById('body');
+    if (!htmlEl) return;
+    if (!bodyEl) return;
+    if (isMenuOpen) {
+      htmlEl.classList.add('overflow-hidden');
+      bodyEl.classList.add('overflow-hidden');
+    } else {
+      htmlEl.classList.remove('overflow-hidden');
+      bodyEl.classList.remove('overflow-hidden');
+    }
+  }, [isMenuOpen]);
```

Comment:
I think a better solution for this could be as follows:

1. On your variables and constants declaration add a ref pointing to the document body: `const bodyRef = useRef<HTMLBodyElement | null>(null);`
2. Replace the highlighted useEffect with these 2 useEffects:
```
  useEffect(() => {
    bodyRef.current = document.querySelector('body');
  }, []);

  useEffect(() => {
    if (bodyRef.current) {
      bodyRef.current.style.overflow = isMenuOpen ? 'hidden' : 'auto';
    }
  }, [isMenuOpen]);
```
This is a cleaner solution for the navbar display while preventing the site scroll.
The first useEffect is to set the ref to the body element and the second one is to mutate the style of the referenced body element to disable the scroll